proxy/http/HttpTransact.cc

```diff
@@ -263,8 +263,13 @@ find_server_and_update_current_info(HttpTransact::State *s)
     TxnDebug("http_trans", "request is from localhost, so bypass parent");
     s->parent_result.result = PARENT_DIRECT;
   } else if (s->method == HTTP_WKSIDX_CONNECT && s->http_config_param->disable_ssl_parenting) {
-    s->parent_params->findParent(&s->request_data, &s->parent_result, s->txn_conf->parent_fail_threshold,
-                                 s->txn_conf->parent_retry_time);
+    if (s->parent_result.result == PARENT_SPECIFIED) {
```

Comment:
Presumably this is initialized to a reasonable value, so the this check is valid despite possibly having to update the value in the `else` class? I wonder if `findParent` should just check its second argument and do the right thing in that case. Would it be too challenging to keep `s->parent_result.result` correct? Although it seems you already have that problem here.